chatkit-core/src/main/kotlin/com/pusher/chatkit/users/UserService.kt

```diff
@@ -26,35 +25,29 @@ class UserService(
                         "id=${URLEncoder.encode(it, "UTF-8")}"
                     }
 
-            client.doGet<List<User>>("/users_by_ids?$missingUserIdsQs")
-                    .map { loadedUsers ->
-                        loadedUsers.map { user ->
-                            knownUsers[user.id] = user
-                        }
+            val fetchResult = client.doGet<List<User>>("/users_by_ids?$missingUserIdsQs")
+            when (fetchResult) {
+                is Result.Success -> {
+                    for (fetchedUser in fetchResult.value) {
+                        knownUsers[fetchedUser.id] = fetchedUser
                     }
+                }
+                is Result.Failure -> return fetchResult.error.asFailure()
```

Comment:
Isn't this reconstructing the same value to return? Does the whole thing reduce to this?

```
            return client.doGet<List<User>>("/users_by_ids?$missingUserIdsQs").map { fetchedUsers ->
                    for (fetchedUser in fetchResult.value) {
                        knownUsers[fetchedUser.id] = fetchedUser

                        presenceService.subscribeToUser(fetchedUser.id)
                    }

                    userIds.map { userId ->
                        userId to knownUsers[userId]!!
                    }
```

Reply:
Cheers @mdpye, [done](a1266dc7210ae45587d8e9be7b2153d9957c9a3c), test's green.